clang/test/SemaCXX/cxx2a-consteval.cpp

```diff
@@ -1032,17 +1043,15 @@ int f() {
 
 namespace GH57682 {
 void test() {
-  constexpr auto l1 = []() consteval { // expected-error {{cannot take address of consteval call operator of '(lambda at}} \
-                                       // expected-note  2{{declared here}}
+  constexpr auto l1 = []() consteval { // expected-note  {{declared here}}
         return 3;
   };
   constexpr int (*f1)(void) = l1; // expected-error {{constexpr variable 'f1' must be initialized by a constant expression}} \
                                   // expected-note  {{pointer to a consteval declaration is not a constant expression}}
 
 
-  constexpr auto lstatic = []() static consteval { // expected-error {{cannot take address of consteval call operator of '(lambda at}} \
-                                       // expected-note  2{{declared here}} \
-                                       // expected-warning {{extension}}
+  constexpr auto lstatic = []() static consteval { // expected-note  {{declared here}} \
+                                                   // expected-warning {{extension}}
```

Comment:
Not that I can tell; I've added one above which assigns the address of the lambda to a non-constexpr variable. The resulting diagnostic points to the `l1` declaration - seems like ideally it would also diagnose at the site of the initializer, but that appears unrelated to this change.

Reply:
I've added some test cases for assigning to a non constexpr variable, both within and outside of an immediate function context.